src/pykeen/models/unimodal/conv_kb.py

```diff
@@ -111,51 +117,13 @@ def __init__(
             ),
             **kwargs,
         )
-
-        self.num_filters = num_filters
-
-        # The interaction model
-        self.conv = nn.Conv2d(in_channels=1, out_channels=num_filters, kernel_size=(1, 3), bias=True)
-        self.relu = nn.ReLU()
-        self.hidden_dropout = nn.Dropout(p=hidden_dropout_rate)
-        self.linear = nn.Linear(embedding_dim * num_filters, 1, bias=True)
-
-    def _reset_parameters_(self):  # noqa: D102
-        # embeddings
-        logger.warning("To be consistent with the paper, initialize entity and relation embeddings from TransE.")
-        super()._reset_parameters_()
-
-        # Use Xavier initialization for weight; bias to zero
-        nn.init.xavier_uniform_(self.linear.weight, gain=nn.init.calculate_gain("relu"))
-        nn.init.zeros_(self.linear.bias)
-
-        # Initialize all filters to [0.1, 0.1, -0.1],
-        #  c.f. https://github.com/daiquocnguyen/ConvKB/blob/master/model.py#L34-L36
-        nn.init.constant_(self.conv.weight[..., :2], 0.1)
-        nn.init.constant_(self.conv.weight[..., 2], -0.1)
-        nn.init.zeros_(self.conv.bias)
-
-    def score_hrt(self, hrt_batch: torch.LongTensor) -> torch.FloatTensor:  # noqa: D102
-        h = self.entity_embeddings(indices=hrt_batch[:, 0])
-        r = self.relation_embeddings(indices=hrt_batch[:, 1])
-        t = self.entity_embeddings(indices=hrt_batch[:, 2])
-
-        # Output layer regularization
+        if regularizer is None:
```

Comment:
how would you turn off regularization?

Reply:
hmm. this wasn't an option in the original model, right? We could put the class name for the regularizer in `__init__` instaed, and if `None` gets passed, it doesn't use it

Reply:
`regularize_if_necessary` should only have done something, if regularization was enabled.

I like your solution, maybe we could also use the class directly instead of the string? (makes it easier to automatically detect if we at one point would rename classes, allows your IDE to directly go to the class, etc.)

Reply:
regularize_if_necessary is not part of the new-style model, though

I'm rethinking this and None is usually interpreted as "use the default". If a user doesn't want regularization, they can pass the no operation regularizer

Reply:
I think we should come back to this in a separate PR